test/functional/vimscript/system_spec.lua

```diff
@@ -691,4 +692,95 @@ describe('shell :!', function()
     feed('<CR>')
     helpers.expect_exit(command, 'qall!')
   end)
+
+  it(':.! should pipe line buffer to &shell only replacing the current line', function()
+    if is_os('win') then
+      -- Output of pwsh is lengthy.
+      -- Do not check the full string which is environment dependent
+      -- but the presence of critical words.
+      helpers.set_shell_powershell(true)
+      insert([[echo 'Hi, coconut!';
+exec zsh;]])
+      feed('gg')
+      command(':.!')
+      local found = string.find(curbuf_contents(), 'Hi, coconut!')
+      neq(found, nil)
```

Comment:
if you want to check partial match use `helpers.matches` .

```suggestion
      helpers.matches('Hi, coconut!', curbuf_contents())
```

Note: use `%` to escape things in Lua patterns. Hyphen is special in Lua patterns so it needs to be escaped to match a literal hypen.

Reply:
Thanks for pointing to this!